### PR TITLE
gdb: update to 11.1

### PR DESCRIPTION
**Pull request by orgads**
Build failures reported upstream:
https://sourceware.org/bugzilla/show_bug.cgi?id=28476

### CODE REVIEW

File: mingw-w64-gdb/PKGBUILD [hunk 1]
```diff
@@ -26,40 +26,47 @@ makedepends=("${MINGW_PACKAGE_PREFIX}-iconv"
              "${MINGW_PACKAGE_PREFIX}-xz")
```
[the remark this answers is not in the document]
Comment (orgads) [reply mingw-w64-gdb/PKGBUILD:27]: Done

**Comment by orgads** [comment]
I think it should compile now. I ran a clean build on my machine to verify, hope it goes well.

**Comment by orgads** [comment]
I had to exclude `erc32` target, because it fails to build with exotic errors, and I have no idea what to do about it.

```
C:\Temp\ccerxL7b.s: Assembler messages:
C:\Temp\ccerxL7b.s:208: Error: value of 0000001000000339 too large for field of 4 bytes at 00000000000000f9
C:\Temp\ccerxL7b.s:307: Error: value of 0000001000000299 too large for field of 4 bytes at 0000000000000191
C:\Temp\ccerxL7b.s:369: Error: value of 000000100000030b too large for field of 4 bytes at 0000000000000203
C:\Temp\ccerxL7b.s:861: Error: value of 000000100000058b too large for field of 4 bytes at 0000000000000483
C:\Temp\ccerxL7b.s:970: Error: value of 0000001000000643 too large for field of 4 bytes at 000000000000053b
C:\Temp\ccerxL7b.s:1077: Error: value of 000000100000077a too large for field of 4 bytes at 0000000000000672
...
```

**Review by jeremyd2019 — APPROVED** [review]
Packaging LGTM, but have not tried it

**Comment by jeremyd2019** [comment]
@GitMensch have you tried the updated packages from CI?

**Comment by GitMensch** [comment]
Hm, for the identifier patch as noted in https://sourceware.org/pipermail/gdb-patches/2021-October/182795.html - you likely want `#ifdef (__USE_MINGW_ANSI_STDIO) && __USE_MINGW_ANSI_STDIO` - because the second will raise at least a warning because it is not guaranteed to be defined in the scope of this patches, is it?

**Comment by orgads** [comment]
No. Any macro that is not defined is considered 0 in `#if`.

See [da docs](https://gcc.gnu.org/onlinedocs/cpp/If.html):

> Identifiers that are not macros, which are all considered to be the number zero. This allows you to write #if MACRO instead of #ifdef MACRO, if you know that MACRO, when defined, will always have a nonzero value. Function-like macros used without their function call parentheses are also treated as zero.

**Pull request merged**